### DISCUSSION TITLE
test_thread_limit_apply FAILED assert statement

### DISCUSSION THREAD

**Discussion by nruloff** [MDAnalysis/mdanalysis#4569] [Installation]
I'm installing MDAnalysis for the first time. The installation is in a Linux environment. Running the code block from the Quick start guide outputs that MDAnalysis is installed and able to successfully run a Universe with 3341 atoms.
![MDAnalysis_Universe_Test](https://github.com/MDAnalysis/mdanalysis/assets/98500424/74cff7f3-9791-44b7-b90c-8640eb79c5be)

I have proceeded to the 'testing' portion of the installation guide, but I am getting a FAILED assert statement from test_base.py 
![command_prompt_test_thread_limit_apply_FAIL](https://github.com/MDAnalysis/mdanalysis/assets/98500424/ff12eb1f-8a85-47f2-83bd-a31c932c71ba)

The code of test_base.py looks like it is performing a transformation to set the max number of threads to 2 prior to the assert statement test by "new_trans = CustomTransformation(max_threads=2).
![test_base_py](https://github.com/MDAnalysis/mdanalysis/assets/98500424/d1ceefb8-61a1-46fd-97b9-8021a0ed7b51)

But the thread info seems to be getting its thread count from the default_thread_info / new_trans is not being impacted by the 'max_threads' input of 'CustomTransformation(max_threads=2).
![default_thread_info_output](https://github.com/MDAnalysis/mdanalysis/assets/98500424/ed4aaacf-7cb8-47f3-b431-07c3ae394ecf)

As I am writing this discussion thread - I am also noticing that the 'testing' portion that I am reading from is a subsection of the installation for 'Development Versions'.

- Does this testing check not apply to my installation as I have not installed a 'Development Version'?
- Does the check from running the code block from the Quick start guide suffice for a successful installation?

Thanks for any help in advance!

**Comment by RMeli** [comment]
Hi @nruloff, it's good you are trying to run all the tests. In principle they should all pass. Looks like you bumped into a new issue: https://github.com/MDAnalysis/mdanalysis/issues/4570

Do you have more informations about your system and environment (specifically, the NumPy version)?

**Reply by nruloff** [reply]
Hi @RMeli - apologies for the delay in reply. The NumPy version that I am using is 1.24.3 (see screenshot below). I can also provide all of the other package versions if that would be helpful. 

The system is a Dell Precision 7865 tower with an AMD Ryzen Threadripper Pro 5995wx CPU (64 cores), two NVIDIA RTX A6000 GPUs and the system is running Ubuntu 22.04.4 LTS for the OS.

![image](https://github.com/MDAnalysis/mdanalysis/assets/98500424/1d5d0884-0739-43c0-9a83-df7af645fc48)

The next plan I have is to run the basic example in a Jupyter environment (as I was able to create a successful Universe of 3341 atoms), and then revisit the Development Version testing. But please let me know any other information which would be helpful for this issue.

**Reply by IAlibay** [reply]
I'm very roughly remembering but I think this might be related to: https://github.com/MDAnalysis/mdanalysis/issues/4570

Could you confirm what version of threadpoolctl you are using @nruloff ?

**Reply by nruloff** [reply]
@IAlibay - certainly. The threadpoolctl version that I am using is 2.2.0 - see the screenshot below.

![image](https://github.com/MDAnalysis/mdanalysis/assets/98500424/4f81ffdc-72b3-4b70-a98a-ff98b994e519)

**Reply by nruloff** [reply]
I should add - the Basic Example test did not show any errors. But I am just starting to read through the Documentation / Tutorials now. If there are any tutorials which specifically test the thread limit, I can test that as well.

**Reply by IAlibay** [reply]
Thanks! That's a very old version of threadpoolctl, ~ 3 years old. I suspect you might need to go for a newer version, maybe trying a new environment with threadpoolctl ~ 3.3+ could fix the issue.

**Reply by nruloff** [reply]
@IAlibay - I updated threadpoolctl to 3.5.0 and reran the pytest and still obtain the same error (screenshot below).

![image](https://github.com/MDAnalysis/mdanalysis/assets/98500424/e51c7a79-ac62-4869-9957-af524ad0f036)


I am noticing that when I run the pytest that there are "7 skipped" - could this potentially be related or is this a separate issue?
